github-actions/DotNet.GitHubAction/DotNet.CodeAnalysis/FileNameUtilities.cs

```diff
@@ -0,0 +1,188 @@
+﻿using System.Diagnostics.CodeAnalysis;
+
+namespace DotNet.CodeAnalysis
+{
+    /// Internals borrowed from:
+    /// https://github.com/dotnet/roslyn/blob/main/src/Compilers/Core/Portable/InternalUtilities/FileNameUtilities.cs
+    /// <summary>
+    /// Implements a few file name utilities that are needed by the compiler.
+    /// In general the compiler is not supposed to understand the format of the paths.
+    /// In rare cases it needs to check if a string is a valid file name or change the extension 
+    /// (embedded resources, netmodules, output name).
+    /// The APIs are intentionally limited to cover just these rare cases. Do not add more APIs.
+    /// </summary>
+    internal static class FileNameUtilities
+    {
+        internal const char DirectorySeparatorChar = '\\';
+        internal const char AltDirectorySeparatorChar = '/';
+        internal const char VolumeSeparatorChar = ':';
+
+        /// <summary>
+        /// Returns true if the string represents an unqualified file name. 
+        /// The name may contain any characters but directory and volume separators.
+        /// </summary>
+        /// <param name="path">Path.</param>
+        /// <returns>
+        /// True if <paramref name="path"/> is a simple file name, false if it is null or includes a directory specification.
+        /// </returns>
+        internal static bool IsFileName([NotNullWhen(returnValue: true)] string? path)
+        {
+            return IndexOfFileName(path) == 0;
+        }
+
+        /// <summary>
+        /// Returns the offset in <paramref name="path"/> where the dot that starts an extension is, or -1 if the path doesn't have an extension.
+        /// </summary>
+        /// <remarks>
+        /// Returns 0 for path ".goo".
+        /// Returns -1 for path "goo.".
+        /// </remarks>
+        private static int IndexOfExtension(string? path)
```

Comment:
I think much of the logic in this method could be simplified using `LastIndexOf`.

Reply:
In a comment, you should make it clear what the result for this method would be in this cases:

- ".bashrc" (any file that starts with "."
- "application.config.json" (any file that contains multiple "." characters.